pkg/formats/elasticsearch/elasticsearch.go

```diff
@@ -0,0 +1,141 @@
+package elasticsearch
+
+import (
+	"bufio"
+	"bytes"
+
+	"github.com/kentik/ktranslate/pkg/formats/util"
+	"github.com/kentik/ktranslate/pkg/kt"
+	"github.com/kentik/ktranslate/pkg/rollup"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/kentik/ktranslate/pkg/eggs/logger"
+)
+
+const (
+	InstNameNetflowEvent = "netflow-events"
+	InstNameVPCEvent     = "vpc-flow-events"
+	InstNameAWSVPCEvent  = "aws-vpc-flow-events"
+	actionEntry          = `{"index":{}}`
+)
+
+var json = jsoniter.ConfigFastest
+
+type ElasticsearchFormat struct {
+	logger.ContextL
+}
+
+func NewFormat(log logger.Underlying) (*ElasticsearchFormat, error) {
+	jf := &ElasticsearchFormat{
+		ContextL: logger.NewContextLFromUnderlying(logger.SContext{S: "elasticsearchFormat"}, log),
+	}
+
+	return jf, nil
+}
+
+func (f *ElasticsearchFormat) To(msgs []*kt.JCHF, serBuf []byte) (*kt.Output, error) {
+	msgsNew := make([]map[string]interface{}, 0, len(msgs))
+	for _, msg := range msgs {
+		if msg.EventType == kt.KENTIK_EVENT_SNMP {
+			continue
+		}
+		mm := msg.Flatten()
+		strip(mm)
+		msgsNew = append(msgsNew, mm)
+	}
+
+	esBulkData := ""
+	for _, m := range msgsNew {
+		data, err := serialize(m)
+		if err != nil {
+			return nil, err
+		}
+		esBulkData += data
```

Comment:
I'm not entirely sure about how go handles string append, but generally faster to make esBulkData a slice of strings and then join with `""` at the end?